### DISCUSSION TITLE
React-native-expo ERROR {"context": "client"} undefined is not a function

### DISCUSSION THREAD

**Discussion by Mustafa6900** [WalletConnect/walletconnect-monorepo#2425] [Developer Support] [answered]
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/58307398/3ead491d-646f-45f4-beef-7c995a91ac21)
expo-crypto-shim.js:
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/58307398/62b44949-c223-4c06-801b-923986c485fe)
When I press the button in this code, I get the following error:

 INFO Cleaning up inactive pairings...
 INFO Inactive pairings cleared: 0
 ERROR {"context": "client"} undefined is not a function
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/58307398/ded42bcd-2fca-41f6-b5f1-13fffb620467)

Steps:

Step 1) https://docs.walletconnect.com/2.0/reactnative/web3modal/installation 
Step 2) npx expo install @web3modal/react-native
Step 3) npx expo install @react-native-async-storage/async-storage react-native-get-random-values react-native-modal react-native-svg
Step 4) npx expo install expo-crypto
Step 5) expo-crypto-shim.js: Adding the file to my project.

**Comment by Mustafa6900** [comment]
Help me

**Comment by finessevanes** [comment]
@ignaciosantise could we get your help 🙏🏽

**Comment by ignaciosantise** [comment, marked as answer]
@Mustafa6900 Did you import the shim file in App.js?

Check this last step: 
https://docs.walletconnect.com/2.0/reactnative/web3modal/Installation#apply-polyfill

**Reply by Mustafa6900** [reply]
There was no js file called shim.js in the project. Should I still import?

**Reply by Mustafa6900** [reply]
I added the expo-crypto-shim.js file as an extra.

**Reply by ignaciosantise** [reply]
@Mustafa6900 thats correct, now import your file "expo-crypto-shim.js" at the top of your App.js

**Reply by Mustafa6900** [reply]
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/58307398/c4200efe-3824-4196-bf97-8ae3132f8cf6)
I added as you said. I started getting this error after pressing the button.

**Reply by ignaciosantise** [reply]
@Mustafa6900 can you move the `import './expo-crypto-shim.js'` line to the top of the file?

**Comment by Mustafa6900** [comment]
Thanks bro. I succeeded.
Thank you for your help and good luck to the walletconnect team.

ignaciosantise ***@***.***>, 29 May 2023 Pzt, 18:17 tarihinde
şunu yazdı:
